bopis-mixed-cart-pos/extensions/pickup-and-delivery-options/src/Pickup.jsx

```diff
@@ -0,0 +1,599 @@
+
+  const api = useApi();
+  const cart = useCartSubscription();
+
+  useEffect(() => {
+    const variantIds = cart.lineItems.map((item) => item.variantId);
+
+    const fetchProductVariants = async () => {
+      const results = await api.productSearch.fetchProductVariantsWithIds(variantIds);
+      const products = results.fetchedResources.reduce((acc, variant) => {
+        acc[variant.id] = variant;
+        return acc;
+      }, {});
+      setProducts(products);
+    };
+
+    if (variantIds.length) {
+      fetchProductVariants();
+    }
+  }, [cart.lineItems]);
+
+  const removeLineItemAttributesSKU = (sku) => {
+    const lineItem = cart.lineItems.find((item) => item.sku === sku);
+    if (lineItem) {
+      api.cart.removeLineItemProperties(lineItem.uuid, ["_pickupstore", "Pick Up", "_delivery_type"]);
+      console.log(`Attributes removed from line item with SKU: ${sku}`);
+    }
+  };
+
+  const addLineItemAttributesSKU = (sku, store) => {
+    const lineItem = cart.lineItems.find((item) => item.sku === sku);
+    if (lineItem) {
+      api.cart.addLineItemProperties(lineItem.uuid, {
+        "_pickupstore": store.value,
+        "Pick Up": store.label,
+        "_delivery_type": "pick_up_instore",
+      });
+      console.log(`Attributes added to line item with SKU: ${sku}`);
+    }
+  };
+
+  const fetchStoresForSelectedSKU = async (sku) => {
+    if (!zipCode) {
+      setZipCodeError("Zip code is required.");
+      return;
+    }
+    if (zipCode.length < 5 || zipCode.length > 9) {
+      setZipCodeError("Zip code length must lie between 5 and 9");
+      return;
+    }
+    setLoading(true);
+    try {
+      const storeList = await getAvailableStoresForSelectedSKU(zipCode, sku);
+      setAvailableStores(storeList);
+      if (storeList.length > 0) {
+        setSelectedStore(storeList[0]);
+      }
+      setShowNoStoresMessage(storeList.length === 0);
+    } catch (error) {
+      console.error("Error fetching stores for SKU:", error);
+    }
+    setLoading(false);
+  };
+
+  const getAvailableStoresForSelectedSKU = async (zipCode, sku) => {
+    try {
+      const postcodeResponse = await fetch(`${API_OMS_BASE_URL}/postcodeLookup`, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          json: {
+            query: `postcode:${zipCode}`,
+          },
+        }),
+      });
+      const postcodeResult = await postcodeResponse.json();
+      const { latitude, longitude } = postcodeResult.response.docs[0];
+      const storeResponse = await fetch(`${API_OMS_BASE_URL}/storeLookup`, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          viewSize: 150,
+          filters: ["storeType: RETAIL_STORE", "pickup_pref: true"],
+          point: `${latitude},${longitude}`,
+          sortBy: "storeName asc",
```

Comment:
No need to send sortBy option, as by default sorting is performed on storeName and is point is passed the sorting is performed on geoDist.